### ISSUE TITLE
Creating an object of a type which is a supertype of the constructor type should not be allowed.

### ISSUE THREAD

**Issue by t-rasmud** [typetools/checker-framework#2187]
Consider the example below:

```
import org.checkerframework.checker.propkey.qual.*;
class ConstructorIssue1 {
    @PropertyKey ConstructorIssue1() {}
}
class ConstructorIssue1Client {
    void test() {
        @UnknownPropertyKey ConstructorIssue1 propObj2 = new @UnknownPropertyKey ConstructorIssue1();
    }
}
```

Since the constructor is annotated as `@PropertyKey`, it should not be allowed to create objects of type `@UnknownPropertyKey`. So the constructor invocation line should report an error. But executing the command `javac -processor propertykey ConstructorIssue1.java` reports no error for this example.

**Comment by t-rasmud** [comment]
The type of a newly created Object can be a supertype of the constructor declaration type. See #2163
